From 731cfcfc640f1c739bca75fd15aeda12e51b2a02 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Fri, 20 Aug 2010 11:16:15 -0700
Subject: Revert "x86: don't send SIGBUS for kernel page faults"
Patch-mainline: never

This reverts commit 495b59364c67bc1f970a6229895ae15d1684d694.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
Automatically created from "patches.kernel.org/revert-x86-don-t-send-sigbus-for-kernel-page-faults.patch" by xen-port-patches.py

--- sle11sp1-2010-08-23.orig/arch/x86/mm/fault-xen.c	2010-08-23 12:10:22.000000000 +0200
+++ sle11sp1-2010-08-23/arch/x86/mm/fault-xen.c	2010-08-23 12:12:16.000000000 +0200
@@ -809,10 +809,8 @@ do_sigbus(struct pt_regs *regs, unsigned
 	up_read(&mm->mmap_sem);
 
 	/* Kernel mode? Handle exceptions or die: */
-	if (!(error_code & PF_USER)) {
+	if (!(error_code & PF_USER))
 		no_context(regs, error_code, address);
-		return;
-	}
 
 	/* User-space => ok to do another page fault: */
 	if (is_prefetch(regs, error_code, address))
